Fix vaccum name in materials

bam241 opened the pull request:
This aims to follow up on #805.

Right now I just pulled @pshriwise branch and rebased it. 

## Description
This fixes the definition of the Graveyard and the Vacuum, to respectively exactly `mat:Graveyard` or `mat:graveyard` and `mat:Vacuum` or `mat:vacuum"`
## Motivation and Context
the previous implementation would lead to detection of a vacuum where there was not, if the name of the material contained "vacuum" or "graveyard"

## Changelog file
- [x] done ?


_This works is sponsored by [Proxima Fusion](https://www.proximafusion.com/)_ 
<img src="https://cdn.prod.website-files.com/65b10cc303e570e609b235c8/65b10cc303e570e609b23618_Asset%205.svg" data-canonical-src="https://www.proximafusion.com" width="100" height="70" /><img src="https://cdn.prod.website-files.com/65b10cc303e570e609b235c8/65b10cc303e570e609b23626_Typo.svg" data-canonical-src="https://www.proximafusion.com" height="70" />

pshriwise committed:
Switching to exact match for vacuum and graveyard checks.

pshriwise committed:
Removing extra print in uwuw test

pshriwise committed:
Using new member variables for comprision and assignment

pshriwise committed:
Adding accessors for material strings and using in mcnp_funcs

bam241 committed:
turning cout into a logged message

bam241 committed:
Add changelog

bam241 commented:
@gonuke in the issue you mention a solution that would be backward compatible with most model out there.

Do you have an idea of the backward compatibility we main encounter ? (so I can implement a fix :) )

gonuke commented:
> @gonuke in the issue you mention a solution that would be backward compatible with most model out there.
> 
> 
> 
> Do you have an idea of the backward compatibility we main encounter ? (so I can implement a fix :) )

Not really... you all are the users 😀 

Maybe it's not reasonable, but I just wanted to make sure it was considered.

bam241 committed:
bug fix

bam241 committed:
formating

bam241 committed:
fixing mcnp bug

bam241 committed:
fixing mcnp bug

bam241 committed:
fixing mcnp bug

bam241 committed:
dagmc_util not dagmc_utils

bam241 committed:
formating

bam241 committed:
using DMD to_lower

bam241 committed:
fix

bam241 committed:
fix

bam241 committed:
fix typo

bam241 committed:
propagating change

bam241 committed:
propagating change

bam241 committed:
removed the wrong import

bam241 committed:
moving to_lower in utils

bam241 committed:
missing import and formating

bam241 committed:
mat_num

bam241 committed:
typo + formating

bam241 committed:
typo + formating

bam241 committed:
back to origin

bam241 committed:
back to origin

bam241 committed:
formating

gonuke commented on src/dagmc/dagmcmetadata.cpp:
```diff
@@ -209,32 +209,30 @@ void dagmcMetaData::parse_material_data() {
-    bool is_vacuum =
-        to_lower(grp_name).find(to_lower(vacuum_str)) != std::string::npos;
+    logger.message("Group name -- " + grp_name);
+    bool is_graveyard = to_lower(grp_name) == to_lower(graveyard_mat_str());
+    bool is_vacuum = to_lower(grp_name) == to_lower(vacuum_mat_str());
```
This may be good for long term readability
```suggestion
    bool is_graveyard = (to_lower(grp_name) == to_lower(graveyard_mat_str()));
    bool is_vacuum = (to_lower(grp_name) == to_lower(vacuum_mat_str()));
```

gonuke commented on src/dagmc/dagmcmetadata.hpp:
```diff
@@ -351,11 +360,13 @@ class dagmcMetaData {
-  const std::string reflecting_str{"Reflecting"};
-  const std::string white_str{"White"};
-  const std::string periodic_str{"Periodic"};
+  const std::string graveyard_str_{"Graveyard"};
+  const std::string vacuum_str_{"Vacuum"};
+  const std::string vacuum_mat_str_{"mat:Vacuum"};
+  const std::string graveyard_mat_str_{"mat:Graveyard"};
+  const std::string reflecting_str_{"Reflecting"};
+  const std::string white_str_{"White"};
+  const std::string periodic_str_{"Periodic"};
```
We frequently - but not always - convert these to lower case.  Is there a reason to not just define them as lower case?

bam241 replied:
I think they are only converted to lower case when comparing with a group name read from a h5 geometry.
My guess is the default if not lower case, but allowing some flexibility for the users.

I don't have preferences there, it is really up to you and Pat :)

@pshriwise as you are the initiator of the branch do you have a take on this ?

gonuke reviewed (changes requested):
Thanks for this cleanup @bam241 .  I have a couple of minor questions, but otherwise it looks good.

If you want to consider this for the next release, you can update the base branch as `v3.2.4-rc1`.  (It may require a rebase on that branch and reconciling conflicts in the ChangeLog.)

bam241 committed:
Apply suggestions from code review

Co-authored-by: Paul Wilson <paul.wilson@wisc.edu>

bam241 commented:
@pshriwise @gonuke would it be interesting to add a setter for the `Graveyard` and `Vacuum` string.

Allowing users to change them on the DAGMC call instead of having to change the whole geometry definition?

That way we could ensure a little of backward compatibility.

we could keep the default as they are now, but allow some flexibility...

bam241 commented:
> @pshriwise @gonuke would it be interesting to add a setter for the `Graveyard` and `Vacuum` string.
> 
> Allowing users to change them on the DAGMC call instead of having to change the whole geometry definition?
> 
> That way we could ensure a little of backward compatibility.
> 
> we could keep the default as they are now, but allow some flexibility...

maybe @makeclean has also an opinion on my last question :)

bam241 committed:
one vacuum mat was not set using vacuum_mat_str method

bam241 committed:
adding setting and tests to check detection of vacuum to avoid regression

bam241 committed:
syntax and comment

bam241 committed:
syntax

bam241 committed:
syntax

bam241 commented:
@gonuke I rebased against `svalinn:v3.2.4-rc1` and changed the target branch.

about your comment, I guess I am not the one with the most expertise about this.
I keep as they were.

bam241 commented:
(btw unit test didn't run after my rebase -- Do they only run on PR vs develop ?.)
